### PR TITLE
Improved compression testing

### PR DESCRIPTION
**Pull request by Christian-B**
Addresses:
https://github.com/SpiNNakerManchester/PACMAN/issues/72

Code currently in a single test may need to be made more generally accessible

### CODE REVIEW

**Comment by alan-stokes** [comment]
all_covers length is never used

File: unittests/operations_tests/router_compressor_tests/test_mundy_compressor.py [hunk 1]
```diff
@@ -8,6 +8,77 @@
...
+            if (not route1.processor_ids == route2.processor_ids):
+                msg = "Value {} was picked up by route: {} in the original " \
+                      "and route: in the compressed.".format(value, route1)
+                raise AssertionError(msg)
+
+    def covers(self, route, length=4):
+        """
+        This method discovers all the routing keys covered by this route
+
+        Starts of wwith the assumption thsat the key is always covered/
```
Comment (alan-stokes) [inline unittests/operations_tests/router_compressor_tests/test_mundy_compressor.py]: typo

File: unittests/operations_tests/router_compressor_tests/test_mundy_compressor.py [hunk 2]
```diff
@@ -8,6 +8,77 @@
...
+        for value in covers:
+            route1 = self.value_checker(original, value)
+            route2 = self.value_checker(compressed, value)
+            if route1 is None:
+                continue
+            if route2 is None:
+                msg = "Value {} was picked up by route: {} in the original " \
+                      "but not in the compressed.".format(value, route1)
+                raise AssertionError(msg)
+            if (not route1.processor_ids == route2.processor_ids):
```
Comment (alan-stokes) [inline unittests/operations_tests/router_compressor_tests/test_mundy_compressor.py]: redundant brackets

**Comment by alan-stokes** [comment]
overall this new version is trying to be too general, but in the end, isn't general enough. 
for this exact test, its fine, but there should be a proper test of the compressor using masks and keys which are more indicative of how we use it in general. masks of ffffffff are not exactly good tests. 

not sure what we want to do with this as it currently stands.

**Comment by Christian-B** [comment]
Leave branch and pull request open for more work.

Neither a blocker for release.

**Comment by Christian-B** [comment]
better solution in https://github.com/SpiNNakerManchester/PACMAN/pull/122

**Pull request closed without merging**